m4/parmetis.m4

```diff
@@ -12,63 +12,79 @@ AC_DEFUN([CONFIGURE_PARMETIS],
                          [AC_MSG_ERROR(bad value ${enableval} for --enable-parmetis)])],
                 [enableparmetis=$enableoptional])
 
+  AC_ARG_WITH(parmetis,
+             AS_HELP_STRING([--with-parmetis=<internal,PETSc,/some/libdir>],
+                            [internal: build from contrib, PETSc: rely on PETSc]),
+             [AS_CASE("${withval}",
+                      [internal], [build_parmetis=yes],
+                      [PETSc],    [build_parmetis=petsc],
+                      [PARMETIS_LIB="-L${withval} -lparmetis"
+                       build_parmetis=no])
+              enableparmetis=yes],
+             [build_parmetis=yes])
+
+  AC_ARG_WITH(parmetis-include,
+             AS_HELP_STRING([--with-parmetis-include=</some/includedir>]),
+             [PARMETIS_INCLUDE="-I${withval}"
+              enableparmetis=yes
+              build_parmetis=no],
+             [])
+
+  dnl Initialize $petsc_have_* to 0 if not already set. They may be
+  dnl unset if the user configured with --disable-petsc
+  AS_IF([test "x$petsc_have_metis" = "x"], [petsc_have_metis=0])
+  AS_IF([test "x$petsc_have_parmetis" = "x"], [petsc_have_parmetis=0])
+
+  dnl If we're using a PETSc with its own METIS, default to using
+  dnl it's ParMETIS too or no ParMETIS at all, regardless of what the
+  dnl user specified (if anything) in --with-parmetis.
+  AS_IF([test $petsc_have_metis -gt 0],
+        [
+         AS_IF([test $petsc_have_parmetis -gt 0],
+               [AC_MSG_RESULT(<<< Using PETSc ParMETIS support to avoid PETSc conflict>>>)
+                build_parmetis=petsc],
+               [AC_MSG_RESULT(<<< Disabling ParMETIS support to avoid PETSc conflict>>>)
+                enableparmetis=no])
+        ])
+
+  dnl Conversely, if:
+  dnl .) ParMETIS is enabled in libmesh,
+  dnl .) PETSc does not have a ParMETIS or METIS, or we aren't using PETSc, and
+  dnl .) build_parmetis=petsc because user said --with-parmetis=PETSc,
+  dnl then we need to make sure that libmesh builds its own ParMETIS!
```

Comment:
Holy moly! In our builds we disable ParMETIS because it has a different (and less LGPL-compatible) license than METIS, so I should be able to pull your branch and test that configuration...

Reply:
Okay, this should be ready for that test.  No rush, though; there's only one system I'm using external Parmetis on right now and I can just rebase branches there on this for the near future.

Reply:
Finally got around to trying this with an `--enable-petsc --with-metis=PETSc --disable-parmetis` build and looks like it works fine. I did not test what I believe is the more challenging case, with system installs of METIS/Parmetis that could conflict with stuff, but it sounds like that is the case which motivated this Issue for you in the first place, so I assume it's working as well...